duckduckgo/Android
# Rewrite tracking links

Task/Issue URL: https://app.asana.com/0/72649045549333/1201353307088098/f

### Description
This PR adds this functionality:

1. Get AMP link formats, AMP keywords and exceptions from remote config.
2. Extract canonical URL from standard AMP links (eg. https://www.google.com/amp/s/www.example.com).
3. Extract canonical URL from cloaked AMP links (eg. https://www.example.com/something.amp).
4. Report initial tracking URL when reporting a broken site.
5. Reference tests for AMP formats and keywords.

### Steps to test this PR
__Link extraction__
1. Navigate to https://privacy-test-pages.glitch.me/privacy-protections/amp/
2. Open the links on the page. The loaded url should match the "Expected" url from the test page.
3. Paste https://www.bbc.com/news/world-asia-china-19432800.amp into the address bar. The loaded page should be https://www.bbc.com/news/world-asia-china-19432800
4. Paste https://www.bbc.com/news/world-asia-china-19432800.amp into another app such as Keep. Open the link with DDG as the default browser. The loaded page should be https://www.bbc.com/news/world-asia-china-19432800

__Broken sites__
1. Filter Logcat with "m_bsr".
2. Navigate to https://privacy-test-pages.glitch.me/privacy-protections/amp/
3. Click one of the "google.com AMP Links".
4. Tap the overflow menu and select "Report Broken Site".
5. Select an option and send the report.
6. Verify that a pixel is sent with the original URL and not the expected value.
7. Go back.
8. Click one of the "First Party Cloaked AMP links".
9. Tap the overflow menu and select "Report Broken Site".
10. Select an option and send the report.
11. Verify that a pixel is sent with the original URL and not the expected value.

__Device Testing__
- [x] Pixel 3 (Android 12)
- [x] OnePlus 8 Pro (Android 11)
- [x] Pixel 2 (Android 11)
- [x] Nexus 6P (Android 8.1)
- [x] Samsung S7 Edge (Android 8)

Some more testing URLs are available here: https://app.asana.com/0/0/1201684971137950/f
